src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp

```diff
@@ -2692,36 +2755,47 @@ class CWixStandardBootstrapperApplication : public CBalBaseBootstrapperApplicati
     }
 
 
-    HRESULT LoadBootstrapperBAFunctions()
+    HRESULT LoadBAFunctions()
     {
         HRESULT hr = S_OK;
         LPWSTR sczBafPath = NULL;
+        BA_FUNCTIONS_CREATE_ARGS bafCreateArgs = { };
+        BA_FUNCTIONS_CREATE_RESULTS bafCreateResults = { };
 
         hr = PathRelativeToModule(&sczBafPath, L"bafunctions.dll", m_hModule);
-        BalExitOnFailure(hr, "Failed to get path to BA function DLL.");
+        BalExitOnFailure(hr, "Failed to get path to BA functions DLL.");
 
 #ifdef DEBUG
-        BalLog(BOOTSTRAPPER_LOG_LEVEL_STANDARD, "WIXSTDBA: LoadBootstrapperBAFunctions() - BA function DLL %ls", sczBafPath);
+        BalLog(BOOTSTRAPPER_LOG_LEVEL_STANDARD, "WIXSTDBA: LoadBAFunctions() - BA functions DLL %ls", sczBafPath);
 #endif
 
         m_hBAFModule = ::LoadLibraryW(sczBafPath);
         if (m_hBAFModule)
         {
-            PFN_BOOTSTRAPPER_BA_FUNCTION_CREATE pfnBAFunctionCreate = reinterpret_cast<PFN_BOOTSTRAPPER_BA_FUNCTION_CREATE>(::GetProcAddress(m_hBAFModule, "CreateBootstrapperBAFunction"));
-            BalExitOnNullWithLastError(pfnBAFunctionCreate, hr, "Failed to get CreateBootstrapperBAFunction entry-point from: %ls", sczBafPath);
+            PFN_BA_FUNCTIONS_CREATE pfnBAFunctionsCreate = reinterpret_cast<PFN_BA_FUNCTIONS_CREATE>(::GetProcAddress(m_hBAFModule, "BAFunctionsCreate"));
+            BalExitOnNullWithLastError(pfnBAFunctionsCreate, hr, "Failed to get BAFunctionsCreate entry-point from: %ls", sczBafPath);
 
-            hr = pfnBAFunctionCreate(m_pEngine, m_hBAFModule, &m_pBAFunction);
-            BalExitOnFailure(hr, "Failed to create BA function.");
+            bafCreateArgs.cbSize = sizeof(bafCreateArgs);
+            bafCreateArgs.qwBAFunctionsAPIVersion = MAKEQWORDVERSION(0, 0, 0, 1); // TODO: need to decide whether to keep this, and if so when to update it.
```

Comment:
You're proposing that `qwBAFunctionsAPIVersion` allows us to drop `cbSize`?  I guess I'm ok with that.  Do the same thing with the BA create args, too?